src/iminuit/minuit.py

```diff
@@ -1043,6 +1051,9 @@ def scipy(
         if ncall is None:
             ncall = self._migrad_maxcall()
 
+        if type(options) is not dict:
+            raise ValueError("options must be a dictionary")
+
```

Comment:
I am not sure whether this is really necessary. One usually does not check types dynamically in Python, unless the type is used to determine what the code does.

Reply:
Type checking is a job for mypy and the linter in your code editor.